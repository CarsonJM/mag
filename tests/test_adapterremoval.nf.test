nextflow_pipeline {

    name "Test pipeline: NFCORE_MAG"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_mag"
    tag "nfcore_mag_test_adapterremoval"

    test("Parameters: clip_tool = 'adapterremoval'") {
        when {
            params {
                input                   = 'https://raw.githubusercontent.com/nf-core/test-datasets/mag/samplesheets/samplesheet.multirun.csv'
                outdir                  = "$outputDir"
                clip_tool               = 'adapterremoval'
                centrifuge_db           = null
                kraken2_db              = null
                keep_phix               = true
                coassemble_group        = true
                skip_spades             = true
                skip_spadeshybrid       = true
                skip_krona              = true
                skip_adapter_trimming   = true
                skip_quast              = true
                skip_prodigal           = true
                skip_binning            = true
                skip_metabat2           = true
                skip_maxbin2            = true
                skip_concoct            = true
                skip_prokka             = true
                skip_binqc              = true
                skip_gtdbtk             = true
                skip_concoct            = true
                megahit_fix_cpu_1       = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    "${outputDir}/Assembly/MEGAHIT/MEGAHIT-group-0.contigs.fa.gz",
                    path("${outputDir}/QC_shortreads/adapterremoval/"),
                    "${params.outdir}/multiqc/multiqc_data/multiqc_fastqc.yaml",
                ).match()},
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() }
            )
        }
    }
}
