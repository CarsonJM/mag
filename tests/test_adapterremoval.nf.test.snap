{
    "Parameters: clip_tool = 'adapterremoval'": {
        "content": [
            "MEGAHIT-group-0.contigs.fa.gz:md5,766fe3571e8774489f9b96ae913386a7",
            [
                [
                    "test_minigut_run1_ar2.settings:md5,eb194399448c3bfe5eced0b1d546a541",
                    "test_minigut_run2_ar2.settings:md5,eb194399448c3bfe5eced0b1d546a541"
                ],
                [
                    "test_minigut_sample2_run0_ar2.settings:md5,8c175a320fab230961c2886e096a019e"
                ]
            ],
            "multiqc_fastqc.yaml:md5,42cf8e13e8ef14ac9192896be672997c"
        ],
        "timestamp": "2023-12-13T00:48:34.419699875"
    }
}